### ISSUE TITLE
Some SPI bits per word values cause kernel panic on BeagleBone Black

### ISSUE THREAD

**Issue by alexanderhiam** [graycatlabs/serbus#1]
On the BeagleBone Black, setting an SPI bus to 11 bits/word then trying to read from it causes kernel panic. Writing works. 

This has something to do with the value because setting it to 16 bits/word works fine for read and write. Perhaps it's odd number values?

**Comment by jpulliam** [comment]
Possibly odd values, happened using 9-bit words. Attached is the log from when I tried to read. Took me forever to track down the problem until I found this bug report.
[capture.txt](https://github.com/graycatlabs/serbus/files/52239/capture.txt)

**Comment by alexanderhiam** [comment]
Yeah, it's an annoying one - are you running into it on a BeagleBone Black? I've been assuming it's an issue in the BeagleBone spi driver, but I haven't tried out odd speeds on any other systems yet.

**Comment by jpulliam** [comment]
Yes, this was on the Beaglebone Black.

edit: just for kicks I ran through a bunch of values:

Working - 4, 5, 6, 7, 8, 12, 13, 14, 15, 16, 32
Broke - 9,10,11,17, 18, 24

**Comment by alexanderhiam** [comment]
Weird... and thanks for testing that! 

Looks like the problem is in here somewhere: https://github.com/beagleboard/linux/blob/3.8/drivers/spi/spi-omap2-mcspi.c#L1034 
Maybe some day I'll look at debugging that driver, but for now I don't think it's too big of a deal to just avoid those broken word sizes - in my experience devices that use weird word sizes are usually compatible with the next biggest power of two anyways.

**Comment by ollie299792458** [comment]
24 bits is quite a common size, see the analog devices ad536X/7X DAC for example
